rtt_roscomm/include/rtt_roscomm/passthrough_callback_queue.hpp

```diff
@@ -8,224 +7,39 @@
-    ops.tracked_object = obj;
-    ops.transport_hints = transport_hints;
-    ops.callback_queue = cq;
-    return nh.subscribe(ops);
-  }
-  template<class M>
-  ros::Subscriber subscribe(
-      ros::NodeHandle& nh, ros::CallbackQueueInterface* cq,
-      const std::string& topic, uint32_t queue_size, void(*fp)(M),
-      const ros::TransportHints& transport_hints = ros::TransportHints())
-  {
-    ros::SubscribeOptions ops;
-    ops.template initByFullCallbackType<M>(topic, queue_size, fp);
-    ops.transport_hints = transport_hints;
-    ops.callback_queue = cq;
-    return nh.subscribe(ops);
-  }
-  template<class M>
-  ros::Subscriber subscribe(
-      ros::NodeHandle& nh, ros::CallbackQueueInterface* cq,
-      const std::string& topic, uint32_t queue_size,
-      void(*fp)(const boost::shared_ptr<M const>&),
-      const ros::TransportHints& transport_hints = ros::TransportHints())
-  {
-    ros::SubscribeOptions ops;
-    ops.template init<M>(topic, queue_size, fp);
-    ops.transport_hints = transport_hints;
-    ops.callback_queue = cq;
-    return nh.subscribe(ops);
-  }
-  template<class M>
-  ros::Subscriber subscribe(
-      ros::NodeHandle& nh, ros::CallbackQueueInterface* cq,
-      const std::string& topic, uint32_t queue_size,
-      const boost::function<void (const boost::shared_ptr<M const>&)>& callback,
-      const ros::VoidConstPtr& tracked_object = ros::VoidConstPtr(),
-      const ros::TransportHints& transport_hints = ros::TransportHints())
-  {
-    ros::SubscribeOptions ops;
-    ops.template init<M>(topic, queue_size, callback);
-    ops.tracked_object = tracked_object;
-    ops.transport_hints = transport_hints;
-    ops.callback_queue = cq;
-    return nh.subscribe(ops);
-  }
-  template<class M, class C>
-  ros::Subscriber subscribe(
-      ros::NodeHandle& nh, ros::CallbackQueueInterface* cq,
-      const std::string& topic, uint32_t queue_size,
-      const boost::function<void (C)>& callback,
-      const ros::VoidConstPtr& tracked_object = ros::VoidConstPtr(),
-      const ros::TransportHints& transport_hints = ros::TransportHints())
-  {
-    ros::SubscribeOptions ops;
-    ops.template initByFullCallbackType<C>(topic, queue_size, callback);
-    ops.tracked_object = tracked_object;
-    ops.transport_hints = transport_hints;
-    ops.callback_queue = cq;
-    return nh.subscribe(ops);
-  }
+class PassthroughCallbackQueue: public ros::CallbackQueueInterface
+{
+  public:
+    /** 
+     * Implementation of ros::CallbackQueueInterface::addCallback()
+     * 
+     * This method is executing the callback received instead of adding
+     * it to a queue. In this way, the queue is bypassed and immediately
+     * executed.
+     * 
+     * @param callback callback to execute, instead of queueing
+     * @param owner_id Owner of the callback, in other implementations it might
+     *                 be used to remove the callback. In this implementation
+     *                 it has no effect.
```

Comment:
```suggestion
     * @param owner_id unused
```

nit: You do not need to explain what other implementations of `ros::CallbackQueueInterface` would probably do with that parameter, or how it is intended to be used by the base class. That is already [documented by roscpp](http://docs.ros.org/api/roscpp/html/classros_1_1CallbackQueueInterface.html) itself.